pkg/cli/server_test.go

```diff
@@ -0,0 +1,138 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"testing"
+
+	"github.com/abcxyz/pkg/cli"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/abcxyz/pkg/testutil"
+)
+
+func TestServerCommand(t *testing.T) {
+	t.Parallel()
+
+	testPrivateKeyString, _ := testGeneratePrivateKey(t)
+
+	ctx := logging.WithLogger(context.Background(), logging.TestLogger(t))
+
+	cases := []struct {
+		name   string
+		args   []string
+		env    map[string]string
+		expErr string
+	}{
+		{
+			name: "success",
+			env: map[string]string{
+				"GITHUB_APP_ID":              "123456",
+				"GITHUB_APP_INSTALLATION_ID": "123456",
+				"GITHUB_APP_PRIVATE_KEY_PEM": testPrivateKeyString,
+			},
+		},
+		{
+			name:   "unexpected args",
+			args:   []string{"foo"},
+			expErr: `unexpected arguments: ["foo"]`,
+		},
+		{
+			name: "invalid_config_missing_github_app_id",
+			env: map[string]string{
+				"GITHUB_APP_INSTALLATION_ID": "123456",
+				"GITHUB_APP_PRIVATE_KEY_PEM": testPrivateKeyString,
+			},
+			expErr: `invalid configuration: GITHUB_APP_ID is empty`,
+		},
+		{
+			name: "invalid_config_missing_github_app_installation_id",
+			env: map[string]string{
+				"GITHUB_APP_ID":              "123456",
+				"GITHUB_APP_PRIVATE_KEY_PEM": testPrivateKeyString,
+			},
+			expErr: `invalid configuration: GITHUB_APP_INSTALLATION_ID is empty`,
+		},
+		{
+			name: "invalid_config_missing_private_key_pem",
+			env: map[string]string{
+				"GITHUB_APP_ID":              "123456",
+				"GITHUB_APP_INSTALLATION_ID": "123456",
+			},
+			expErr: `invalid configuration: GITHUB_APP_PRIVATE_KEY_PEM is empty`,
+		},
+		{
+			name: "invalid_config_invalid_private_key_pem",
+			env: map[string]string{
+				"GITHUB_APP_ID":              "123456",
+				"GITHUB_APP_INSTALLATION_ID": "123456",
+				"GITHUB_APP_PRIVATE_KEY_PEM": "invalid_pem",
+			},
+			expErr: `failed to decode PEM formated key`,
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx, done := context.WithCancel(ctx)
+			defer done()
+
+			var cmd ServerCommand
+			cmd.SetLookupEnv(cli.MultiLookuper(
+				cli.MapLookuper(tc.env),
+			))
+
+			_, _, _ = cmd.Pipe()
+
+			_, err := cmd.RunUnstarted(ctx, tc.args)
+			if diff := testutil.DiffErrString(err, tc.expErr); diff != "" {
+				t.Fatal(diff)
+			}
+			if err != nil {
+				return
+			}
+		})
+	}
+}
+
+// testGeneratePrivateKey generates a rsa Key for testing use.
+func testGeneratePrivateKey(tb testing.TB) (string, *rsa.PrivateKey) {
```

Comment:
This is the same function in the `plugin` package, will do a refactor later and put this in `internal/testhelper/pem.go`

Reply:
maybe in `pkg/plugin/testutil/private_key.go`

Reply:
How about `pkg/testutils`?

Reply:
if you are only using in inside plugin, then pkg/plugin/testutil is better.